src/Util/Datastore.php

```diff
@@ -67,14 +67,16 @@ public function getFile(string $filePath): array
     /**
      * Moves an uploaded file to datastore disk location.
      *
-     * @param array $fileStream File stream resource object.
+     * @param array  $fileStream      File stream resource object.
+     * @param string $destinationPath File destination path on datastore.
      *
      * @return string
      */
-    public function addFile(array $fileStream): string
+    public function addFile(array $fileStream, string $destinationPath): string
     {
-        $uuid = Uuid::uuid4();
-        $destinationPath = self::FILES_DIRECTORY . DIRECTORY_SEPARATOR . $uuid->toString();
+        $uuid = Uuid::uuid4()->toString();
+        // add only last 5 bytes of uuid to the destination path
+        $destinationPath = $destinationPath . '_' . substr($uuid, -5);
```

Comment:
```suggestion
        $destinationPath .= '_' . substr($uuid, -5);
```
Here you could use the "Concat Assignment Operator"